Users/abehenderson/CSX/Closures/once.js

```diff
@@ -0,0 +1,55 @@
+// Attempt #2 (CSX Solution)
+
+function once(cb) {
+  let hasBeenCalled = false;
+  let result;
+  
+  function oncified(...args) { // Uses 'rest operator' to accept an array if needed.  
+
+    if (!hasBeenCalled) {
+      result = cb(...args); // Uses 'spread operator' to invoke 'cb' on all elements on an array, if an array is passed in (must be for an edge case, but it wasn't specified in instructions that we needed to account for an array)
+      hasBeenCalled = true;
+    }
+    return result;
+  }
+  return oncified;
+}
+
+  // Is using 'rest' and 'spread operators' considered best case scenario in a situation like this?  If this was a tech interview, I would just ask edge case questions and find out what the interviewer wants me to account for.
+
+//----------------------------------------------------------------------------------------
+// Attempt #1
+
+function once(cb) {
+  let result;
+  return function(n) {
+    if (result === undefined) { // Is there any case where using 'result' as defined or undefined may fail, vs using a boolean as in the CSX example above?  Is one preferred over the other, or is there a best practice?
```

Comment:
In this case, because this is an abstract heuristic, it's hard to say. A boolean value will catch `undefined` since it is `falsy`, so I'd opt for that instead.